components/image-builder-mk3/pkg/resolve/resolve_test.go

```diff
@@ -20,14 +20,16 @@ import (
 	"github.com/gitpod-io/gitpod/image-builder/pkg/resolve"
 	"github.com/golang/mock/gomock"
 	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
 	"github.com/opencontainers/go-digest"
 	ociv1 "github.com/opencontainers/image-spec/specs-go/v1"
 )
 
 func TestStandaloneRefResolverResolve(t *testing.T) {
 	type Expectation struct {
-		Ref   string
-		Error string
+		Ref          string
+		Error        string
+		ErrorMatcher func(error) bool
```

Comment:
It seems uintuitive too me that the ErrorMatch is part of the Expectation. I would have expected it in the test case definition i.e. L 41

Reply:
> It seems uintuitive too me that the ErrorMatcher is part of the Expectation

I somewhat agree: I would greatly prefer to specify an "positive" `Error string` we can simply compare with. But I'm lagging knowledge as to how to construct it in a way that it :100: resembles the error we get from `containerd`.
If you have any ideas, please let me know

Given that, making it part of the `Expectation` makes sense as it overrides the `Error` check: it's just another way to check whether reality matches our expectation.